Sources/Filters/Sources/PlatonicSolidSource/index.js

```diff
@@ -0,0 +1,197 @@
+    ],
+    cells: [
+      0, 3, 5, 1, 5, 3, 1, 9, 2, 1, 2, 8, 0, 11, 7, 0, 7, 10, 2, 4, 6, 7, 6, 4,
+      3, 10, 9, 4, 9, 10, 5, 8, 11, 6, 11, 8, 1, 3, 9, 1, 8, 5, 0, 10, 3, 0, 5,
+      11, 7, 4, 10, 7, 11, 6, 2, 9, 4, 2, 6, 8,
+    ],
+    numPoints: 12,
+    cellSize: 3,
+    numCells: 20,
+    scale: 1.0 / 0.58778524999243,
+  },
+
+  dodecahedron: {
+    points: [
+      0.38196601125, 0, 1, -0.38196601125, 0, 1, 0.38196601125, 0, -1,
+      -0.38196601125, 0, -1, 0, 1, -0.38196601125, 0, 1, 0.38196601125, 0, -1,
+      -0.38196601125, 0, -1, 0.38196601125, 1, 0.38196601125, 0, 1,
+      -0.38196601125, 0, -1, 0.38196601125, 0, -1, -0.38196601125, 0,
+      -0.61803398875, 0.61803398875, 0.61803398875, 0.61803398875,
+      -0.61803398875, 0.61803398875, -0.61803398875, -0.61803398875,
+      -0.61803398875, 0.61803398875, 0.61803398875, -0.61803398875,
+      0.61803398875, 0.61803398875, 0.61803398875, -0.61803398875,
+      0.61803398875, -0.61803398875, -0.61803398875, -0.61803398875,
+      0.61803398875, 0.61803398875, -0.61803398875, -0.61803398875,
+    ],
+    cells: [
+      0, 16, 5, 12, 1, 1, 18, 7, 13, 0, 2, 19, 6, 14, 3, 3, 17, 4, 15, 2, 4, 5,
+      16, 8, 15, 5, 4, 17, 10, 12, 6, 7, 18, 11, 14, 7, 6, 19, 9, 13, 8, 16, 0,
+      13, 9, 9, 19, 2, 15, 8, 10, 17, 3, 14, 11, 11, 18, 1, 12, 10,
+    ],
+    numPoints: 20,
+    cellSize: 5,
+    numCells: 12,
+    scale: 1.0 / 1.070466269319,
+  },
+};
+
+// ----------------------------------------------------------------------------
+// vtkPlatonicSolidSource methods
+// ----------------------------------------------------------------------------
+
+function vtkPlatonicSolidSource(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkPlatonicSolidSource');
+
+  publicAPI.requestData = (inData, outData) => {
+    const output = outData[0] || vtkPolyData.newInstance();
+
+    let solidData;
+    switch (model.solidType) {
+      case SolidType.VTK_SOLID_TETRAHEDRON:
+        solidData = geometries.tetrahedron;
+        break;
+      case SolidType.VTK_SOLID_CUBE:
+        solidData = geometries.cube;
+        break;
+      case SolidType.VTK_SOLID_OCTAHEDRON:
+        solidData = geometries.octahedron;
+        break;
+      case SolidType.VTK_SOLID_ICOSAHEDRON:
+        solidData = geometries.icosahedron;
+        break;
+      case SolidType.VTK_SOLID_DODECAHEDRON:
+        solidData = geometries.dodecahedron;
+        break;
+      default:
+        solidData = geometries.tetrahedron;
+        break;
+    }
+
+    let pointType;
+    if (model.outputPointsPrecision === DesiredOutputPrecision.SINGLE) {
+      pointType = VtkDataTypes.FLOAT;
+    } else if (model.outputPointsPrecision === DesiredOutputPrecision.DOUBLE) {
+      pointType = VtkDataTypes.DOUBLE;
+    }
+
+    const points = vtkPoints.newInstance({
+      dataType: pointType,
+      numberOfPoints: solidData.numPoints,
+    });
+
+    const colors = vtkDataArray.newInstance({
```

Comment:
this color array is useless.

When that code was written originally, there was no way to easily get the IDs of each cell. This a changed since then (e.g. `vtkIdFilter`)